File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    
    answer = 1 #자기자신
    for i in range(1,(n//2)+1):
        cnt = 0
        for j in range(i,(n//2)+2):
            cnt += j
            if cnt == n:
                answer+=1
                break
            elif cnt > n:
                break
    
    return answer

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
import java.util.*;
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        LinkedList<Integer[]> arr = new LinkedList<>();
        
        for(int i=1; i*(i-1)/2<n;i++){ 
            if((n-(i*(i-1)/2))%i==0){
                answer++;
            }
            
                
            }
        
        
        
        
        return answer;
    }
}